web/src/sidebar_ui.ts

```diff
@@ -209,15 +209,30 @@ export function initialize(): void {
 }
 
 export function initialize_left_sidebar(): void {
+    const is_all_messages_home_view =
+        user_settings.web_home_view === settings_config.web_home_view_values.all_messages.code;
+    const is_recent_view_home_view =
+        user_settings.web_home_view === settings_config.web_home_view_values.recent_topics.code;
+
+    // default order
+    let navigation_list_order = ["inbox", "recent", "feed"];
```

Comment:
This seems really brittle to me, in part because of the other arrays that would have to be adjusted should we, say, need to add another home view to the navigation list. Would it be possible to write some logic to just pull the home view code out of the array, and then stick it onto the front?

Reply:
So you mean dynamically determine the current home view `<li>` element and order it to be the first element in the DOM ?

Reply:
Take a look at [this file](https://github.com/zulip/zulip/pull/31873/files#diff-12b2069984da6a06da644a85394f1160b60a83f873f6683fcdd91839e89ad548), in my previous PR.

Is it what you mean ?